app/controllers/parks_controller.rb

```diff
@@ -43,12 +43,14 @@ def update
 
   def destroy
     @park = Park.find(params[:id])
+    @park.destroy
+
     if @park.destroyed?
-      flash[:notice] = "You failed to delete park!"
-      render :show
-    else
       flash[:notice] = "You have deleted park successfully!"
       redirect_to root_path
+    else
+      flash[:notice] = "You failed to delete park!"
+      render :show
     end
   end
 
```

Comment:
We now have `@park = Park.find(params[:id])` in four places in this controller.. Time to abstract it out:

```
class ParksController < ApplicationController
  ...

  def show
    park
    @review = Review.new
  end

  def edit
    park
  end

  def update
    if park.update(park_params)
      flash[:notice] = "You have successfully updated your question!"
      redirect_to park
    else
      flash.now[:notice] = "Invalid input."
      render 'edit'
    end
  end

  def destroy
    if park.destroy
      flash[:notice] = "You have deleted park successfully!"
      redirect_to root_path
    else
      flash[:notice] = "You failed to delete park!"
      render :show
    end
  end

  private

  def park
    @park ||= Park.find(params[:id])
  end

  def park_params
    ...
  end
end
```

Notice with this strategy that the first line in both `update` and `destroy` takes care of (1) finding the Park object by calling `park` method, (2) assigning it to an instance variable accessed by calling `park` method, (3) performing an action on it, and (4) checking to see if the action performed properly.

That's the advantage of using `update` - it calls `save` for us, so we don't need to explicitly save it afterward, and it also returns true or false depending on whether it saved or not (ie whether the operation successfully performed). Same with `destroy` - it returns true if it executed successfully.